src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1156,6 +1153,21 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Inserts several vertices.
+   * @param[in] vertices A range of Vertex_handle
+   * @param[in] filt filtration value of the new vertices (the same for all)
+   *
+   * This may be faster than inserting the vertices one by one, especially in a random order.
+   * The complex does not need to be empty before calling this function. However, if a vertex is
+   * already present, its filtration value is not modified, unlike with other insertion functions. */
+  template <class VertexRange>
+  void insert_batch_vertices(VertexRange const& vertices, Filtration_value filt = 0) {
+    auto verts = vertices | boost::adaptors::transformed([&](auto v){
+        return Dit_value_t(v, Node(&root_, filt)); });
```

Comment:
If we do not want to capture everything, here we could do:
```suggestion
    auto verts = vertices(skel_graph) | boost::adaptors::transformed([&root = this->root_, &skel_graph](auto v){
        return Dit_value_t(v, Node(&root, get(vertex_filtration_t(), skel_graph, v))); });
```
Maybe not that interesting...

Reply:
Is this comment meant for the function insert_graph instead?

Reply:
"not capture everything" -> "be explicit about what we capture"?
I checked, and the lambda in insert_graph already has exactly the size of 2 pointers, this wouldn't change it.
Ah, it would capture `this->root_` instead of `this`, is that what you meant?
Just trying to understand the goal before I change something.

Reply:
Yes, sorry, I mixed your 2 changes with `boost::adaptors::transformed`.
And yes my comment was to know whether we should be explicit about what we capture in the lambda, or capture everything by reference.
Here, if we capture only 2 pointers, I am ok with it.